### PR TITLE
make Widget/Integration mgr optional (#3224)

### LINKED ISSUE #3224: Make sure there are no ugly edge cases running Riot without an integrations manager
Edge cases we need to manage (display an error or hide the feature):

- integration manager screen
- Sticker picker
- jitsi conf call

iOS issue: https://github.com/vector-im/riot-ios/issues/2518

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3224 

Make widget/integration manager optional.
Integration manager is read from preferences with default values to new vector integration server.
Integration (Matrix app / stickers) ui elements are removed when no integration manager is defined

image

image

### CODE REVIEW

File: vector/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -809,6 +809,7 @@
     <string name="settings_logged_in">Logged in as</string>
     <string name="settings_home_server">Home Server</string>
     <string name="settings_identity_server">Identity Server</string>
+    <string name="settings_integration_server">Integration Server</string>
```
Comment (manuroe) [inline vector/src/main/res/values/strings.xml]: "Integration Manager"

File: vector/src/main/res/values/config.xml [hunk 2]
```diff
@@ -14,8 +14,7 @@
     <string name="bug_report_url" translatable="false">https://riot.im/bugreports/submit</string>
 
     <!-- Widget urls -->
-    <string name="integrations_ui_url" translatable="false">"https://scalar.vector.im/"</string>
-    <string name="integrations_rest_url" translatable="false">"https://scalar.vector.im/api"</string>
+    <string name="integration_server_url" translatable="false">"https://scalar.vector.im"</string>
```
Comment (manuroe) [inline vector/src/main/res/values/config.xml]: I would prefer that we keep 2 app settings for the moment. One for the UI. One for the API.

File: vector/src/main/java/im/vector/widgets/WidgetManagerProvider.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        val serverURL = PreferencesManager.getIntegrationServerUrl(context)
+        if (serverURL.isNullOrBlank()) return null
+        return try {
+            URL(serverURL)
+            val defaultWhitelist = ArrayList(PreferencesManager.getIntegrationWhiteListedUrl(context))
+            defaultWhitelist.add(0,"$serverURL/api")
+
+            val config = IntegrationManagerConfig(
+                    uiUrl = "$serverURL/",
+                    apiUrl = "$serverURL/api",
```
Comment (manuroe) [inline vector/src/main/java/im/vector/widgets/WidgetManagerProvider.kt]: That seems a bit dangerous atm. `/api` suffix is not guaranteed.
This is something that will change but not before phase:2

File: vector/src/main/res/values/config.xml (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/res/values/config.xml]: Ok, changed to use 3 settings (added integrations_jitsi_widget_url)

File: vector/src/main/java/im/vector/widgets/WidgetManagerProvider.kt (hunk 3 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/widgets/WidgetManagerProvider.kt]: changed

File: vector/src/main/res/values/strings.xml (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/res/values/strings.xml]: done

**Pull request merged**